### PR TITLE
feat(compress): add snappy

### PR DESCRIPTION
**Pull request by xkx9431**
### What problem does this PR solve?

Issue Number: close/fix/resolve/ref #775 

### What is changed and how it works?

1.  move the writer / reader pool into lib
2.  add the snappy and lz4 compression method
3.
Please describe how it works

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test A
- [ ] Test B
- [ ] Test cases to be added
- [ ] No code

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by xkx9431** [comment]
@vicky-run  @xuthus5 please help review

File: lib/util/lifted/influx/httpd/handler_compress.go [hunk 1]
```diff
@@ -242,3 +261,77 @@ func putZstdWriter(zstdEncoder *zstd.Encoder) {
 }
 
 // ******************** endregion zstd write pool ***********************
+
+// ******************** region snappy write pool ***************************
+type SnappyWriterPool struct {
```
Comment (fx408) [inline lib/util/lifted/influx/httpd/handler_compress.go:266]: 这一系列的对象池，建议统一放到 lib/pool 下（或者其他更好的地方）
后续还有其他功能，应该也会用到，放在这里就不适合了

**Comment by xiangyu5632** [comment]
@xkx9431 检视意见处理了吗

File: lib/compress/compression_pool.go [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+
+func PutSnappyWriter(snappyWriter *snappy.Writer) {
+	snappyWriter.Close()
+	snappyWriterPool.Put(snappyWriter)
+}
+
+// ******************** endregion snappy write pool ***************************
+
+// ******************** region lz4 write pool ***************************
+type Lz4WriterPool struct {
```
Comment (fx408) [inline lib/compress/compression_pool.go:201]: 在不同类型对象做 pool 的时候，整体带来逻辑是一致的，可以使用泛型进行收编

- 重新写一个通用的pool池。使用泛型进行改造
```
type FixedPoolV2[T any] struct {
	pool chan T
	new  func() T
}

func NewFixedPoolV2[T any](new func() T) *FixedPoolV2[T] {
	return &FixedPoolV2[T]{
		pool: make(chan T, 10),
		new:  new,
	}
}

func (p *FixedPoolV2[T]) Get() T {
	select {
	case item := <-p.pool:
		return item
	default:
		return p.new()
	}
}
```

- 使用上，跟进需求建立不同对象的实体，
- 大幅减少重复代码，去掉了interface{} 到各种类型的转换过程
```
// 全局变量实例化对象池
var snappyPool = pool.NewFixedPoolV2[*snappy.Writer](func() *snappy.Writer {
    return &snappy.Writer{}
})

// 全局变量实例化对象池
var lz4Pool = pool.NewFixedPoolV2[*lz4.Writer](func() *lz4.Writer {
    return &lz4.Writer{}
})

func TestFixPoolV2(t *testing.T) {
	var snappyWriter *snappy.Writer // 先定义变量类型
	snappyWriter = snappyPool.Get()  // 从对应的对象池中获取对象

	var lz4Writer *lz4.Writer
	lz4Writer = lz4Pool.Get()
}
```

Comment (xkx9431) [reply lib/compress/compression_pool.go]: Resolved, but with this method the lz4 still failed in typecheck.

Check source code, the `lz4.NewWriter()` is different with others, so remove lz4 first.

Comment (xkx9431) [reply lib/compress/compression_pool.go]: Will check the lz4 later

**Review by fx408 — APPROVED** [review]

**Pull request merged**